### PR TITLE
wallet export: Derive a private key for mnemonic accounts

### PR DESCRIPTION
**Pull request by matevz**
Implements #51.

For user:
- `oasis wallet show` also prints wallet kind (e.g. `file (ed25519-adr8:3)`
- `oasis wallet export`, if the account is of mnemonic type, also derives the corresponding private key and shows it. This is useful for dApp developers where they need a specific private key to sign the contract create/call transactions

Internal refactorings:
- test account factory was added for consistency with `file` and `ledger` kinds.
- copies of `accountConfig` used by `wallet/file/file.go`, `wallet/ledger/ledger.go` and `wallet/test/test.go` were consolidated into shared `AccountConfig` and moved into `wallet`.

Before:
```
$ oasis wallet export TestEd
WARNING: Exporting the account will expose secret key material!
Unlock your account.
? Passphrase: 
Name:             TestEd
Public Key:       RWAfdhrxfbpQJDUp5ilzLxxY0I/92qhJEjhUBHVynYU=
Native address:   oasis1qqjkrr643qv7yzem6g4m8rrtceh42n46usfscpcf
Export:
equip will roof matter pink blind book anxiety banner elbow sun young

$ oasis wallet export TestSecp
WARNING: Exporting the account will expose secret key material!
Unlock your account.
? Passphrase: 
Name:             TestEth
Public Key:       A4MYU1tUEF1Keq5gwI/EX5aHGBtP38YlvRp1P6c5f+11
Ethereum address: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
Native address:   oasis1qpwyqlu69ga9nzz0xsm0y5j2lr3l52280gdqpkzf
Export:
test test test test test test test test test test test junk

$ oasis wallet export TestSr
WARNING: Exporting the account will expose secret key material!
Unlock your account.
? Passphrase: 
Name:             TestSr
Public Key:       skCcMfU7D0olbxZtn8lLn0u41rJzxiCw7rcv3uVEw3Y=
Native address:   oasis1qz3sz6s86k9l3hlk78s25x29esn5qq43rvrkrgey
Export:
equip will roof matter pink blind book anxiety banner elbow sun young

$ oasis wallet export test:alice
WARNING: Exporting the account will expose secret key material!
Name:             test:alice
Public Key:       NcPzNW3YU2T+ugNUtUWtoQnRvbOL9dYSaBfbjHLP1pE=
Native address:   oasis1qrec770vrek0a9a5lcrv0zvt22504k68svq7kzve
Export:
OPYg5hO22BvHgHmtjo/dZgyn/+efqDhcAfcZ6NsMKhc1w/M1bdhTZP66A1S1Ra2hCdG9s4v11hJoF9uMcs/WkQ==
```

After:
```
$ oasis wallet export TestEd
WARNING: Exporting the account will expose secret key material!
Unlock your account.
? Passphrase: 
Name:             TestEd
Kind:             file (ed25519-adr8:0)
Public Key:       RWAfdhrxfbpQJDUp5ilzLxxY0I/92qhJEjhUBHVynYU=
Native address:   oasis1qqjkrr643qv7yzem6g4m8rrtceh42n46usfscpcf
Secret mnemonic:
equip will roof matter pink blind book anxiety banner elbow sun young
Derived secret key for account number 0:
TpyhpMLtkMkNqT6hgVV++fRl9ETAt9412ushj5OQ2YVFYB92GvF9ulAkNSnmKXMvHFjQj/3aqEkSOFQEdXKdhQ==

$ oasis wallet export TestSecp
WARNING: Exporting the account will expose secret key material!
Unlock your account.
? Passphrase: 
Name:             TestSecp
Kind:             file (secp256k1-bip44:0)
Public Key:       A4MYU1tUEF1Keq5gwI/EX5aHGBtP38YlvRp1P6c5f+11
Ethereum address: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
Native address:   oasis1qpwyqlu69ga9nzz0xsm0y5j2lr3l52280gdqpkzf
Secret mnemonic:
test test test test test test test test test test test junk
Derived secret key for account number 0:
ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80

$ oasis wallet export TestSr
WARNING: Exporting the account will expose secret key material!
Unlock your account.
? Passphrase: 
Name:             TestSr
Kind:             file (sr25519-adr8:3)
Public Key:       skCcMfU7D0olbxZtn8lLn0u41rJzxiCw7rcv3uVEw3Y=
Native address:   oasis1qz3sz6s86k9l3hlk78s25x29esn5qq43rvrkrgey
Secret mnemonic:
equip will roof matter pink blind book anxiety banner elbow sun young
Derived secret key for account number 3:
zxVO/G3spH1IAlRbuUFgqaUR3Zv8N+1FblnYvQ0tewq/dI+++j/jvEQ7Ym9wlRVjgHwKW1IOjJpKqgkN+7IyjQ==

$ oasis wallet export test:alice
WARNING: Exporting the account will expose secret key material!
Name:             test:alice
Kind:             test (ed25519-raw)
Public Key:       NcPzNW3YU2T+ugNUtUWtoQnRvbOL9dYSaBfbjHLP1pE=
Native address:   oasis1qrec770vrek0a9a5lcrv0zvt22504k68svq7kzve
Secret key:
OPYg5hO22BvHgHmtjo/dZgyn/+efqDhcAfcZ6NsMKhc1w/M1bdhTZP66A1S1Ra2hCdG9s4v11hJoF9uMcs/WkQ==
```

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -130,8 +148,8 @@ type Account interface {
 	// SignatureAddressSpec returns the signature address specification associated with the account.
 	SignatureAddressSpec() types.SignatureAddressSpec
 
-	// UnsafeExport exports the account's secret state.
-	UnsafeExport() string
+	// UnsafeExport reveals the account's private key and mnemonic.
```
Comment (kostko) [inline wallet/wallet.go:151]:
```suggestion
	// UnsafeExport returns the account's private key and mnemonic.
```

File: wallet/wallet.go [hunk 2]
```diff
@@ -32,6 +33,23 @@ const (
 	AlgorithmSr25519Raw = "sr25519-raw"
 )
 
+type AccountConfig struct {
```
Comment (kostko) [inline wallet/wallet.go:36]: Add documentation comment.

File: wallet/wallet.go [hunk 3]
```diff
@@ -32,6 +33,23 @@ const (
...
+	Algorithm string `mapstructure:"algorithm"`
+	Number    uint32 `mapstructure:"number,omitempty"`
+}
+
+// Unmarshal imports the config map to AccountConfig.
+func (af *AccountConfig) Unmarshal(raw map[string]interface{}) error {
```
Comment (kostko) [inline wallet/wallet.go:42]:
```suggestion
func (af *AccountConfig) UnmarshalMap(raw map[string]interface{}) error {
```

File: wallet/file/sr25519_test.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package file
+package file //nolint: dupl
```
Comment (kostko) [inline wallet/file/sr25519_test.go:1]: Why is this needed?

File: wallet/file/secp256k1.go [hunk 5]
```diff
@@ -20,21 +20,21 @@ const (
...
-		return nil, fmt.Errorf("failed to derive key from mnemonic: %w", err)
+		return nil, nil, fmt.Errorf("failed to derive key from mnemonic: %w", err)
 	}
 	pk, err := wallet.PrivateKeyBytes(account)
 	if err != nil {
-		return nil, fmt.Errorf("failed to obtain generated private key: %w", err)
+		return nil, nil, fmt.Errorf("failed to obtain generated private key: %w", err)
 	}
-	return secp256k1.NewSigner(pk), nil
+	return secp256k1.NewSigner(pk), pk, nil
```
Comment (kostko) [inline wallet/file/secp256k1.go:37]: Uh, can we rename this to `sk` so it is not confused with public key.

File: wallet/file/sr25519_test.go (hunk 4 above)
Comment (matevz) [reply wallet/file/sr25519_test.go:1]: Linter complains that ed25519_test.go and sr25519_test.go are too similar although they aren't (different private key for the same mnemonic).

**Review by kostko — APPROVED** [review]
